Common/src/main/java/mezz/jei/common/transfer/BasicRecipeTransferHandlerServer.java

```diff
@@ -281,9 +283,14 @@ private static Set<Slot> merge(Map<Slot, ItemStack> result, Map<Slot, ItemStack>
 	}
 
 	@Nullable
-	private static Slot getSlotWithStack(Player player, ItemStack stack, List<Slot> craftingSlots, List<Slot> inventorySlots) {
+	private static Slot getSlotWithStack(Player player, ItemStack stack, List<Slot> craftingSlots, List<Slot> inventorySlots, Slot hint) {
 		Slot slot = getSlotWithStack(player, craftingSlots, stack);
+
 		if (slot == null) {
+			if (!hint.getItem().isEmpty() && ItemStack.isSameItemSameTags(stack, hint.getItem()) && hint.mayPickup(player)) {
```

Comment:
just a style thing but I like to split up long lines like this so that there is one method call per line
```suggestion
                        Item hintItem = hint.getItem();
			if (!hintItem.isEmpty() && 
			    ItemStack.isSameItemSameTags(stack, hintItem) && 
			    hint.mayPickup(player)) {
```